source/elements/oneTBB/source/algorithms/blocked_ranges/blocked_rangeNd_cls.rst

```diff
@@ -0,0 +1,175 @@
+                using size_type = typename dim_range_type::size_type;
+
+                // Constructors
+                blocked_rangeNd(const dim_range_type& dim0 /*, ... - exactly N parameters of the same type*/);
+                blocked_rangeNd(value_type size[N], size_type grainsize = 1);
+                blocked_rangeNd(blocked_rangeNd& r, split); 
+                blocked_rangeNd(blocked_rangeNd& r, proportional_split proportion); 
+
+                // Capacity
+                static constexpr unsigned int dim_count();
+                bool empty() const;
+
+                // Access
+                bool is_divisible() const;
+                const dim_range_type& dim(unsigned int dimension) const;
+            };
+
+        } // namespace tbb
+    } // namespace oneapi        
+
+Requirements:
+
+* ``N`` must be greater than 0.
+* The *Value* must meet the :doc:`BlockedRangeValue requirements <../../named_requirements/algorithms/blocked_range_val>`.
+
+Member types
+------------
+
+.. code:: cpp
+
+    using value_type = Value;
+
+The type of the range values.
+
+.. code:: cpp
+
+    using dim_range_type = blocked_range<value_type>;
+
+The type that represents one out of the N dimensions.
+
+.. code:: cpp
+
+    using size_type = typename dim_range_type::size_type;
+
+The type for measuring the size of a dimension.
+
+Member functions
+----------------
+
+.. code:: cpp
+
+    blocked_rangeNd( const dim_range_type& dim0 /*, ... - exactly N parameters of the same type*/ );
+
+**Effects:**  Constructs a ``blocked_rangeNd`` representing an N-dimensional space of values.
+The space is the half-open Cartesian product of one-dimensional ranges ``dim0 x ...``.
+The constructor must take exactly N arguments, which types match ``const dim_range_type&``.
+
+**Example:** For ``blocked_rangeNd<int,4>``, this constructor is equivalent to
+``blocked_rangeNd( const blocked_range<int>&, const blocked_range<int>&, const blocked_range<int>&, const blocked_range<int>& )``.
+
+.. note::
+    This constructor cannot be substituted with a variadic template constructor
+    ``template <typename... Dims> blocked_rangeNd( const Dims&... dims )``, even if the latter
+    is constrained by the size and type requirements for the parameter pack ``Dims``.
+    That is because the types in ``Dims`` could not be automatically deduced from arguments specified as
+    braced initialization lists, and so expressions like ``blocked_rangeNd<int, 4>{{0,1},{0,2},{0,3},{0,4}}``
+    would fail to compile.
+
+.. code:: cpp
+
+    blocked_rangeNd( value_type size[N], size_type grainsize = 1 );
+
+**Effects:**  Constructs a ``blocked_rangeNd`` representing an N-dimensional space of values.
+The space is the half-open Cartesian product of ranges ``[0, size[0]) x [0, size[1]) x ...``
+each having the same grain size.
+
+**Example:**  The ``blocked_rangeNd<int,4> r( {5,6,7,8}, 4 );`` statement constructs a four-dimensional
```

Comment:
All the other ranges constructors gives you ability to specify begins of the ranges. 
At least from my perspective it less confusing to have both begins and ends explicitly specified.
Otherwise, `blocked_rangeNd<int,2> r( {1,6} );` could be confused into `blocked_rangeNd<int,2> r( {/*begin*/1,/*end*/6} );`.

Reply:
We discussed it here https://github.com/uxlfoundation/oneAPI-spec/pull/555#discussion_r1730432024 and agreed to only provide the "simplest" version that takes the array of sizes.

> Otherwise, `blocked_rangeNd<int,2> r( {1,6} );` could be confused into `blocked_rangeNd<int,2> r( {/*begin*/1,/*end*/6} );`.

`blocked_rangeNd<int,2> r( {/*begin*/1,/*end*/6} );.` should have a second dimension though to be meaningful, no?

Reply:
>should have a second dimension though to be meaningful, no?

Also true but still looks a bit confusing 😄

Reply:
@pavelkumbrasev should we consider this conversation "resolved", or do you view it as an objection that needs to be addressed in the design?

Reply:
"Resolved"